llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -3442,4 +3442,11 @@ def int_amdgcn_addrspacecast_nonnull : DefaultAttrsIntrinsic<
   [llvm_anyptr_ty], [llvm_anyptr_ty],
   [IntrNoMem, IntrSpeculatable]
 >;
+
+/// Make it clear to the backend that this value is really dead. For instance,
```

Comment:
So is this just convergent poison? Why isn't it IntrConvergent?

Reply:
Because in my current usage it's ok for it to be sunk from the entry block into `shader`. In my very limited understanding, convergent would preclude that (but do correct me if I'm wrong).

Reply:
But is hosting correct? In the future with convergence tokens this will be explicit for where movement can occur